Topics/Development_Process/Build_Requirements/CMakeLists_txt.md

```diff
@@ -0,0 +1,229 @@
+
+Unlike Make, which is a build system that drives the compiler and other build tools to build your code, CMake (or Cross-platform Make) is a "generator" of build systems that can be used for general purpose builds. It is much more high-level such that it can generate a low-level build script in Ninja or Make or many other build systems that you can run with.
+
+The key feature of CMake is that it supports the cross-platform discovery of system libraries, and automatic discovery and configuration of the toolchain, which is much easier to use than Make.
+
+To install CMake, check their [document](https://cmake.org/download/) on how to download and install on Unix/Linux or Windows.
+
+
+## What is CMakeLists.txt?
+
+When CMake processes a project, the entry point is a source file called `CMakeLists.txt` in the top level of the source directory.
+
+`CMakeLists.txt` is the CMake configuration file, which contains a set of directives and instructions describing the project's source files and targets (executable, library, or both) that will be used to build the program. It determines everything for the building process, from which source files to compile, to which options to choose for the libraries and to present to the users.
+
+Here is a simple example of how it looks like:
+```CMake
+# CMakeLists.txt
+
+# The minimum version of CMake required
+cmake_minimum_required(VERSION 3.10)
+
+# Project name
+project(MyProject)
+
+# Add main.cpp file of the project root directory as a source file
+set(SOURCE_FILES main.cpp)
+
+# Add executable targets
+add_executable(MyExecutable ${SOURCE_FILES})
+
+```
+
+## Creating CMakeLists.txt
+
+### CMake Language
+`CMakeLists.txt` is written in [CMake Language](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#manual:cmake-language(7)), which consists of comments, [commands](https://cmake.org/cmake/help/latest/manual/cmake-commands.7.html#manual:cmake-commands(7)), and [variables](https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html#manual:cmake-variables(7)). 
+
+* **comments** start with `#`
+  ```CMake
+  # This is a comment
+  ```
+* **commands** start with command names and a list of whitespace-separated arguments (case-sensitive).
+  ```CMake
+  find_package(LIBIGL REQUIRED QUIET)
+  ```
+* **variables** are case-sensitive with only alphanumeric characters and underscores. 
+
+  A number of useful variables are automatically defined by CMake (e.g `CMAKE_CURRENT_SOURCE_DIR`, `PROJECT_NAME`). 
+
+  Use *set* command to set variable names, and *${variable_name}* to reference a varible in command arguments. 
+  ```CMake
+  set(Foo a b c)    # Value of Foo is "a;b;c"
+
+  command(${Foo})   # Foo is replaced by a;b;c 
+                    # and expands to 3 arguments
+  ```
+
+It also provides the basic **control flow** - Conditional statements, Looping constructs, and Procedure definitions. Here are the examples:
```

Comment:
Weird mid-sentence capitalization here as well -- and start-of-bullet-point non-capitalization below